my-app/src/components/CardList/CardList.jsx

```diff
@@ -0,0 +1,31 @@
+import Card from '../Card/Card.jsx';
+import WithLoadingDelay from '../Loading/WithLoadingDelay.jsx';
+import '../../App.css';
+import { SongContext } from '../../app_context/song-context.jsx';
+import { useContext } from 'react';
+import PropTypes from 'prop-types';
+
+const CardWithLoading = WithLoadingDelay(Card);
+
+export default function CardList({ viewOnly }) {
+  const { songs, getSelectedCards } = useContext(SongContext);
+
+  return (
+    <div className="cards">
+      {songs.map(songCard => (
+        <CardWithLoading
+          key={songCard.id}
+          {...songCard}
+          viewOnly={viewOnly}
+          getSelectedCards={isChecked => {
```

Comment:
No need to pass `getSelectedCards` as a prop - you can get it from the context in the Card component. Also, I would recommend to rename it - the name `updateSelectedCard` is more in line with what the function really does.